SpatialGDK/Source/SpatialGDK/Private/Tests/RPCServiceTest.cpp

```diff
@@ -496,3 +508,128 @@ RPC_SERVICE_TEST(GIVEN_receiving_an_rpc_WHEN_return_false_from_extract_callback_
 	TestTrue("Returning false in extraction callback correctly stopped processing RPCs", bTestPassed);
 	return true;
 }
+
+DEFINE_LATENT_AUTOMATION_COMMAND_ONE_PARAMETER(FWaitForWorld, TSharedPtr<TestData>, Data);
+bool FWaitForWorld::Update()
+{
+	UWorld* World = nullptr;
+	const TIndirectArray<FWorldContext>& WorldContexts = GEngine->GetWorldContexts();
+	for (const FWorldContext& Context : WorldContexts)
+	{
+		if ((Context.WorldType == EWorldType::PIE || Context.WorldType == EWorldType::Game)
+			&& (Context.World() != nullptr))
+		{
+			World = Context.World();
+			break;
+		}
+	}
+
+	if (World != nullptr && World->AreActorsInitialized())
+	{
+		AGameStateBase* GameState = World->GetGameState();
+		if (GameState != nullptr && GameState->HasMatchStarted())
+		{
+			Data->TestWorld = World;
+			return true;
+		}
+	}
+
+	return false;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_ONE_PARAMETER(FSpawnActor, TSharedPtr<TestData>, Data);
+bool FSpawnActor::Update()
+{
+	FActorSpawnParameters SpawnParams;
+	SpawnParams.bNoFail = true;
+	SpawnParams.SpawnCollisionHandlingOverride = ESpawnActorCollisionHandlingMethod::AlwaysSpawn;
+
+	AActor* Actor = Data->TestWorld->SpawnActor<ATestActor>(SpawnParams);
+	Data->Actor = Actor;
+
+	return true;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_ONE_PARAMETER(FWaitForActor, TSharedPtr<TestData>, Data);
+bool FWaitForActor::Update()
+{
+	AActor* Actor = Data->Actor;
+	return (IsValid(Actor) && Actor->IsActorInitialized() && Actor->HasActorBegunPlay());
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_TWO_PARAMETER(FDropRPCQueueTest, FAutomationTestBase*, Test, TSharedPtr<TestData>, Data);
+bool FDropRPCQueueTest::Update()
```

Comment:
Can you refactor this down into separate steps rather than the whole test in one?

Reply:
Is that necessary? The reason these are separated is for temporal reasons no?

Reply:
Yea fair, I guess not in this case, I think if we want to add more tests then more modular steps would help a lot but can delay for now if you pref